pkgs/desktops/gnome/misc/gnome-flashback/default.nix

```diff
@@ -155,15 +155,38 @@ let
             '';
           };
 
-          executable = writeShellScriptBin "gnome-flashback-${wmName}" ''
-            if [ -z $XDG_CURRENT_DESKTOP ]; then
-              export XDG_CURRENT_DESKTOP="GNOME-Flashback:GNOME"
-            fi
-
-            export XDG_DATA_DIRS=${wmApplication}/share:${gnomeSession}/share:${gnome-flashback}/share:${gnome-panel}/share:$XDG_DATA_DIRS
+          # gnome-panel will only look for applets in a single directory so symlink them into here.
+          panelModulesEnv = buildEnv {
+            name = "panel-modules-env";
+            # We always want to find the built-in panel applets.
+            paths = [ gnome-panel ] ++ panelModulePackages;
+            pathsToLink = [ "/lib/gnome-panel/modules" ];
+          };
 
-            exec ${gnome-session}/bin/gnome-session --session=gnome-flashback-${wmName} "$@"
-          '';
+          executable = stdenv.mkDerivation {
+            name = "gnome-flashback-${wmName}";
+            nativeBuildInputs = [ wrapGAppsHook ];
+            buildInputs = [ glib gnome-panel gnome.gnome-flashback ] ++ panelModulePackages;
```

Comment:
Does this need to be `gnome.gnome-flashback` rather than just `gnome-flashback`. if so could you add a comment explaining that?